[CI] SnapshotMetricsIT testSnapshotAPMMetrics failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12250 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/e7t7r5yhkofiw)
- [elasticsearch-pull-request #145906 / part-1](https://gradle-enterprise.elastic.co/s/sp5l7fr3pmjdy)
- [elasticsearch-pull-request #145641 / checkpart1_release-tests](https://gradle-enterprise.elastic.co/s/fw2xhoqjvjjqa)
- [elasticsearch-pull-request #145305 / part-1](https://gradle-enterprise.elastic.co/s/r5bpfgchmkxa2)

**Reproduction Line:**
```
gradlew ":server:internalClusterTest" --tests "org.elasticsearch.repositories.SnapshotMetricsIT.testSnapshotAPMMetrics" -Dtests.seed=4BCCA83F52C71F05 -Dtests.locale=cs-Latn-CZ -Dtests.timezone=Asia/Manila -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.repositories.SnapshotMetricsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSnapshotAPMMetrics'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: every item is a value less than <0.8066585>
     but: an item <0.807> was greater than <0.8066585>
```

**Issue Reasons:**
- [main] 4 failures in test testSnapshotAPMMetrics (0.5% fail rate in 853 executions)
- [main] 2 failures in step part-1 (0.4% fail rate in 519 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (0.6% fail rate in 520 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -548,9 +548,6 @@ tests:
 - class: org.elasticsearch.index.codec.vectors.diskbbq.ES920DiskBBQVectorsFormatTests
   method: testOneRepeatedVector
   issue: https://github.com/elastic/elasticsearch/issues/149254
-- class: org.elasticsearch.repositories.SnapshotMetricsIT
-  method: testSnapshotAPMMetrics
-  issue: https://github.com/elastic/elasticsearch/issues/149256
 - class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardingPitSearchIT
   method: testPitRelocationDuringReshard
   issue: https://github.com/elastic/elasticsearch/issues/149271

--- a/server/src/internalClusterTest/java/org/elasticsearch/repositories/SnapshotMetricsIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/repositories/SnapshotMetricsIT.java
@@ -112,7 +112,7 @@ public class SnapshotMetricsIT extends AbstractSnapshotIntegTestCase {
         // Block the snapshot to test "snapshot shards in progress"
         blockAllDataNodes(repositoryName);
         final String snapshotName = randomIdentifier();
-        final long beforeCreateSnapshotNanos = System.nanoTime();
+        final long beforeCreateSnapshotMs = System.currentTimeMillis();
         final ActionFuture<CreateSnapshotResponse> snapshotFuture;
         try {
             snapshotFuture = clusterAdmin().prepareCreateSnapshot(TEST_REQUEST_TIMEOUT, repositoryName, snapshotName)
@@ -136,7 +136,7 @@ public class SnapshotMetricsIT extends AbstractSnapshotIntegTestCase {
 
         // wait for snapshot to finish to test the other metrics
         safeGet(snapshotFuture);
-        final TimeValue snapshotElapsedTime = TimeValue.timeValueNanos(System.nanoTime() - beforeCreateSnapshotNanos);
+        final TimeValue snapshotElapsedTime = TimeValue.timeValueMillis(System.currentTimeMillis() - beforeCreateSnapshotMs);
         final double maxHistogramDurationSeconds = snapshotElapsedTime.secondsFrac() + DOUBLE_HISTOGRAM_DURATION_UPPER_BOUND_SLACK_SEC;
         collectMetrics();
 

